src/cpp/server_core/ServerDatabase.cpp

```diff
@@ -151,7 +151,23 @@ Error readOptions(const std::string& databaseConfigFile,
          // always ensure the database file user is correct, if specified
          error = databaseFile.changeOwnership(databaseFileUser.get());
          if (error)
+         {
+            error.addProperty("description", "Unable to change ownership of database file");
             return error;
+         }
+
+#ifndef _WIN32
+         error = databaseDirectory.changeOwnership(databaseFileUser.get());
```

Comment:
on non windows platforms, this line is duplicated (on successful tests) Should we move the `bool writable = false ...` block inside the first `if (error)` block and avoid calling it twice?

Reply:
The first changeOwnership call is for databaseFile. This one is for databaseDirectory... due to some weird sqlite requirement that the directory itself be writable.

Reply:
Ah! Such close names! Makes sense to me!